### PR TITLE
Schema evolution procedural specs

### PR DESCRIPTION
**Pull request by ESapenaVentura**
- Added schema evolution procedural specs document under the `docs` folder for review

### CODE REVIEW

File: docs/dcp2_schema_evolution_specs.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+The HCA metadata standard is developed in a transparent and open manner so that the whole HCA community can participate
+in the process. It follows a set of `design principles`_ that serve as a foundation for the metadata standard in order
+to ensure it meets the needs of HCA data contributors, consumers and DCP downstream components.
+
+The metadata schemas are versioned following semantic `versioning rules`_ that account for backwards compatibility.
+However, compatibility of the schemas does not account for impact on the schema consumers, especially if different
+versions of the same schemas must be maintained, indexed, displayed and/or interacted with in any form.
+
+The metadata is updated through PRs in the `metadata-schema repository`_. The specific details can be found in the
+documentation in the repository, but the overall flow can be described as follows:
```
Comment (hannes-ucsc) [inline docs/dcp2_schema_evolution_specs.rst:25]:
```suggestion
```

File: docs/dcp2_schema_evolution_specs.rst [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+- A ticket is opened describing an issue
```
Comment (hannes-ucsc) [inline docs/dcp2_schema_evolution_specs.rst:27]: where?

File: docs/dcp2_schema_evolution_specs.rst [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+- A member of the Metadata Team looks into the issue, there is a discussion and, if deemed important, a PR is created
```
Comment (hannes-ucsc) [inline docs/dcp2_schema_evolution_specs.rst:28]: Define "Metadata Team"

File: docs/dcp2_schema_evolution_specs.rst [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  against the schema
+- The PR is reviewed within a time frame specified by the type of change (Major/Minor/Patch). The review includes
```
Comment (hannes-ucsc) [inline docs/dcp2_schema_evolution_specs.rst:30]: This Major/Minor/Patch distinction is dangerously simplistic. For one, a logical schema change could affect different schema documents in different ways, and bump different parts of their semantic version numbers. More importantly, a minor, or even patch level change to a schema document could have major impacts downstream. You simply don't know when you make the change.

File: docs/dcp2_schema_evolution_specs.rst (hunk 1 above)
Comment (hannes-ucsc) [reply docs/dcp2_schema_evolution_specs.rst:25]: It's ok to link to other texts but it's not ok to "conceptually include" texts via link that you think are normative. A spec needs to be self-contained as much as possible.

File: docs/dcp2_schema_evolution_specs.rst [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+  all the DCP components and test metadata is generated for each PR.
```
Comment (hannes-ucsc) [inline docs/dcp2_schema_evolution_specs.rst:31]: This needs to be much more specific. It should specify that
- the test data over time will exercise every graph shape found in the wild
- the test metadata is kept up to date with the latest schema
- the document versions and identifiers are stable
- the test metadata has minimal duplication but maximal diversity
- the data files are dummyes of zero length

File: docs/dcp2_schema_evolution_specs.rst [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+- Once the required metadata team reviewers approve the changes, the branch is merged into staging
```
Comment (hannes-ucsc) [inline docs/dcp2_schema_evolution_specs.rst:32]: To unspecific. Both the test metadata and the schema PRs need to be approved for either of them to be merged.

File: docs/dcp2_schema_evolution_specs.rst [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+- Make it simpler to collect metadata for HCA projects
+- Reduce complexity of the schema towards users
+- Increase clarity of the schema towards users
+- Increase the trust the community has in our ability to build standards
+- Users get an accurate representation of the “full” amount of HCA data
+- Support Managed Access Data
+- Support Contributor Generated Matrices: Representation, linking.
+- Allow new single cell techniques in the DCP
+
+
```
Comment (hannes-ucsc) [inline docs/dcp2_schema_evolution_specs.rst:55]:
```suggestion
```

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
I wanted to give this an honest review pass but I had to stop. This needs to be much more limited in scope. It should elide goals, motivations and other fluff. It should just describe the procedure. And it should not contain redundant texts. You describe various concerns repeatedly, at different levels of detail. Doing so is counter-productive in a specification.

I propose that you whittle this down significantly. Think of it as a sequence of steps, a flow chart in textual form. It needs to be self-contained, concise, precise, and stripped of redundancy. It will be much shorter in the end and easier to review.

This will be a considerable effort, more like a rewrite. If you don't want to do it, I understand, in which case I would author an alternative PR.

Last but not least: Wrap on 80 columns, please. And please rename this document to `docs/operating_procedures.rst` and assume that it will contain multiple SOPs, yours being the first. The section heading hierarchy will change accordingly. The H1 heading at the top should read "HCA DCP Standard Operating Procedures".

File: docs/dcp2_schema_evolution_specs.rst (hunk 4 above)
Comment (mshadbolt) [reply docs/dcp2_schema_evolution_specs.rst:30]: Do you have a suggestion for a better way to categorise/version changes? As you point out, the metadata team doesn't have in-depth knowledge about how the changes we make affect DCP components or other consumers of our schema. The best we can do is version our schema based on how the schema itself changes and the compatibility between schema versions. We don't have control over how other DCP components or independent users of our schema interpret and make assumptions about the schema so I am not sure how we can factor these things into how we version the schema? Or are you saying the way we apply the major/minor/patch categorisations needs to be reviewed?

Comment (hannes-ucsc) [reply docs/dcp2_schema_evolution_specs.rst:30]: We can continue to version the individual schemas as we do right now, I'm not objecting to that. I'm objecting to the assumption that a minor schema change has minor impact on downstream components. There is no easy way (and no need) to categorize schema changes by impact *before* the PR is made. The determination of impact is what the PR review process is all about.

I'm working on a less verbose and more precise version of this SOP which will make that clear.

**Comment by hannes-ucsc** [comment]
Closing in favor or #14

**Pull request closed without merging**